exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusCollector.java

```diff
@@ -5,36 +5,60 @@
 
 package io.opentelemetry.exporter.prometheus;
 
+import io.opentelemetry.sdk.common.CompletableResultCode;
 import io.opentelemetry.sdk.metrics.data.MetricData;
 import io.opentelemetry.sdk.metrics.export.MetricProducer;
+import io.opentelemetry.sdk.metrics.export.MetricReader;
 import io.prometheus.client.Collector;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 
-public final class PrometheusCollector extends Collector {
-  private final MetricProducer metricProducer;
+public final class PrometheusCollector extends Collector
+    implements MetricReader, MetricReader.Factory {
+  private MetricProducer metricProducer;
```

Comment:
`volatile` just to be sure?

Reply:
Shouldn't be necessary as this will be filled out before the class is registered w/ prometheus.  

We can add it, or I can document the reason it's ok not to be volatile here.

Reply:
Added documentation on why, PTAL and let me know if you'd still like volatile here.

Reply:
Hm. I read your description and I'm not convinced. :) The JMM is a tricky beast and visibility of non-final, non-volatiles is likely to trip one up at the worst possible time (in production). Is "shared with other threads" a concrete JMM thing for non-final, non-volatiles?